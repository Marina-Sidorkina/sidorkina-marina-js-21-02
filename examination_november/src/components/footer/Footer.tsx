import React from 'react';
import './Footer.scss';
import Theme from '../theme/Theme';

const Footer = () => (
  <footer className="footer">
    <div className="footer__container">
      <div className="footer__copyright">Delta World &copy; 1970-2077</div>
      <Theme />
    </div>
  </footer>
);

export default Footer;
