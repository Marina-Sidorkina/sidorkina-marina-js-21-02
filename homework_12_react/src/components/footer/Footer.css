.footer {
  box-sizing: border-box;
  position: absolute;
  bottom: 0;
  width: 1024px;
  height: 75px;
  padding-left: 15px;
  padding-top: 25px;
  border-top: 5px solid darkmagenta;
  background-color: white;
  opacity: 1;
}

.footer__text {
  margin-top: 0;
  margin-bottom: 0;
}
