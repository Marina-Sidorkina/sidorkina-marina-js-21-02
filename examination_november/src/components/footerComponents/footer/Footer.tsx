import React, { useContext } from 'react';
import './Footer.scss';
import Theme from '../theme/Theme';
import { ThemeContext } from '../../../contexts/ThemeContext';

const Footer = () => {
  const themeContext = useContext(ThemeContext);

  return (
    <footer className={`${themeContext.darkTheme ? 'footer footer_dark' : 'footer'}`}>
      <div className="footer__container">
        <div className="footer__copyright">Delta World &copy; 1970-2077</div>
        <Theme />
      </div>
    </footer>
  );
};

export default Footer;
