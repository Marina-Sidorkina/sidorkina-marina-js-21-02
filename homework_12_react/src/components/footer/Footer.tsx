import React from "react";
import "./Footer.css";

class Footer extends React.Component {
  render() {
    return (
      <footer className="footer">
        <p className="footer__text">Marina Sidorkina 2021</p>
      </footer>
    )
  }
}

export default Footer;
