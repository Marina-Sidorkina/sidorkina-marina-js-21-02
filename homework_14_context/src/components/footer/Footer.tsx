import React from "react";
import "./Footer.css";
import Theme from "../theme/Theme";
import Paginator from "../paginator/Paginator";
import {IFooterProps} from "../../@types/interfaces/components";

class Footer extends  React.Component<IFooterProps> {
  render() {
    return (
      <footer className="footer">
        <Paginator onCurrentChange={ this.props.onPageChange }
                   current={ this.props.currentPage }
                   className={ this.props.paginatorClassName }/>
        <Theme className={ this.props.themeClassName }/>
      </footer>
    );
  }
}

export default Footer;
