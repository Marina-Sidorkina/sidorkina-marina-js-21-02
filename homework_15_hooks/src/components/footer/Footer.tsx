import React from "react";
import "./Footer.scss";
import Theme from "../theme/Theme";
import Paginator from "../paginator/Paginator";
import {IFooterProps} from "../../@types/interfaces/components";

const Footer = (props: IFooterProps) => {
  return (
    <footer className="footer">
      <Paginator onCurrentChange={ props.onPageChange }
                 current={ props.currentPage }
                 className={ props.paginatorClassName }/>
      <Theme className={ props.themeClassName }/>
    </footer>
  );
}

export default Footer;
