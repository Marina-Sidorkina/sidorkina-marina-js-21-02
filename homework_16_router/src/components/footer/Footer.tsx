import React from "react";
import "./Footer.scss";
import Theme from "../theme/Theme";
import Paginator from "../paginator/Paginator";
import { IFooterProps } from "../../@types/interfaces/components";

const Footer = (props: IFooterProps) => {
  return (
    <footer className="footer">
      {!props.showPaginator ? null :
        <Paginator onCurrentChange={ props.onPageChange }
                   current={ props.currentPage }
                   itemsAmount={ props.itemsAmount }
                   perPageLimit={ props.perPageLimit }/>
      }
      <Theme />
    </footer>
  );
}

export default Footer;
