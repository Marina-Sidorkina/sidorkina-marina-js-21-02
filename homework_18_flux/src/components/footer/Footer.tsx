import React from "react";
import "./Footer.scss";
import Theme from "../theme/Theme";
import Paginator from "../paginator/Paginator";
import { IFooterProps } from "../../@types/interfaces/components";

const Footer = (props: IFooterProps) => {
  const { onPageChange, currentPage, perPageLimit, showPaginator, itemsAmount} = props;
  return (
    <footer className="footer">
      {!showPaginator ? null :
        <Paginator onCurrentChange={ onPageChange }
                   current={ currentPage }
                   itemsAmount={ itemsAmount }
                   perPageLimit={ perPageLimit }/>
      }
      <Theme />
    </footer>
  );
}

export default Footer;
