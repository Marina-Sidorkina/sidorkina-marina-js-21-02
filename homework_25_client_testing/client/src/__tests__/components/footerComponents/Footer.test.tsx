import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import Footer from '../../../components/footerComponents/footer/Footer';

configure({ adapter: new Adapter() });

describe('Footer component test', () => {
  test('should render Main', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.length).toEqual(1);
  });

  test('should render Footer', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
