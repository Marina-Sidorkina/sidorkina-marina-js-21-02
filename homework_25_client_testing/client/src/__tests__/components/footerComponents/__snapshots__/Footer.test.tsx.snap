// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component test should render Footer 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="copyright"
    >
      Delta World Â© 1970-2077
    </div>
    <LanguageSelect />
    <Theme />
  </div>
</footer>
`;
